core/src/main/java/org/apache/druid/java/util/common/JodaUtils.java

```diff
@@ -182,17 +185,16 @@ private static void verifyAscendingSortOrder(Interval previous, Interval current
 
   public static Interval umbrellaInterval(Iterable<Interval> intervals)
   {
-    ArrayList<DateTime> startDates = new ArrayList<>();
-    ArrayList<DateTime> endDates = new ArrayList<>();
+    DateTimeComparator dateTimeComp = DateTimeComparator.getInstance();
+
+    DateTime minStart = new DateTime(Long.MAX_VALUE);
+    DateTime maxEnd = new DateTime(Long.MIN_VALUE);
 
     for (Interval interval : intervals) {
-      startDates.add(interval.getStart());
-      endDates.add(interval.getEnd());
+      minStart = Collections.min(ImmutableList.of(minStart, interval.getStart()), dateTimeComp);
+      maxEnd = Collections.max(ImmutableList.of(maxEnd, interval.getEnd()), dateTimeComp);
     }
 
-    DateTime minStart = minDateTime(startDates.toArray(new DateTime[0]));
-    DateTime maxEnd = maxDateTime(endDates.toArray(new DateTime[0]));
-
     if (minStart == null || maxEnd == null) {
```

Comment:
This function now will return an interval of `Long.MIN_VALUE, Long.MAX_VALUE` when the input `intervals` is empty. It should throw an exception instead.

Reply:
BTW, I'm OK if you don't want to make this change in this PR. It's optional and up to you.

Reply:
Oops. Fixed.